### PR TITLE
Problem: zconfig chunk parser fails reading 'long string' values

### PR DESCRIPTION
**Pull request by FrancoisRegisDegott-eaton**
zconfig_chunk_load() fixed
* handle cur_line dyn. buffer
Signed-off-by: Degott, Francois Regis <FrancoisRegisDegott@eaton.com>

### CODE REVIEW

File: src/zconfig.c [hunk 1]
```diff
@@ -607,9 +610,23 @@ zconfig_chunk_load (zchunk_t *chunk)
         else
             cur_size = remaining;
 
-        if (cur_size > 1024)
-            cur_size = 1024;
-        char cur_line [1024 + 1];
+        if (cur_line && (cur_line != cur_line_p))
+            free (cur_line); //  Release prev. allocated cur_line
+
+        if (cur_size > 1024) {
```
Comment (boricj) [inline src/zconfig.c:616]: Would be better if constant is linked to size of buffer.

File: src/zconfig.c [hunk 2]
```diff
@@ -939,6 +961,47 @@ zconfig_print (zconfig_t *self)
...
+{
+    if (verbose) printf("\n%s\n", "zconfig_test_save_load_long_value");
+
+    zconfig_t *root = zconfig_new ("root", NULL);
+    assert (root);
+    zconfig_t *item = zconfig_new ("item", root);
+    assert (item);
+
+    // see zconfig_chunk_load(), predefined char *cur_line[1024 + 1];
+    const int sz0 = 1000, sz1 = 1048;
```
Comment (boricj) [inline src/zconfig.c]: Would be better if constants are linked to buffer size in `zconfig_chuck_load()`.

**Review by boricj — APPROVED** [review]
Looks good enough.

**Comment by perrettecl** [comment]
Dpo not forget to propose the patch for upstream as we would like to stop to use forks.

**Review by perrettecl — APPROVED** [review]

**Comment by FrancoisRegisDegott-eaton** [comment]
> 
> 
> Dpo not forget to propose the patch for upstream as we would like to stop to use forks.

https://github.com/zeromq/czmq/pull/2082

**Comment by sphaero** [comment]
it's merged upstream

**Pull request merged**